sktime/classification/base.py

```diff
@@ -170,6 +170,7 @@ def fit(self, X, y):
 
         self.classes_ = np.unique(y)
         self.n_classes_ = self.classes_.shape[0]
+        self._X_metadata = X_metadata
```

Comment:
Probably want to add this to _init_ if we are writing to it in every fit.

Reply:
doesn't work in `__init__`, because back then we won't have seen any data yet

Reply:
we dont know the classes prior to fit, but we have self.classes_ = [] in __init__

Reply:
unless self._X_metadata is used anywhere prior to fit which i am missing

Reply:
what precisely are you suggesting? `X_metadata` is available earliest in `fit`, so we cannot write it in `__init__`

Reply:
`self._X_metadata = []`? 

This is what happens for every other variable we write to in fit im pretty sure

Reply:
can add this, but it wouldn't change anything, sure, if that's convention

Reply:
I think its more just to indicate that it exists and will be written to prior to calling fit, very minor but it is convention currently.

Reply:
done